daffodil-runtime1/src/main/scala/org/apache/daffodil/infoset/InfosetWalker.scala

```diff
@@ -452,7 +466,7 @@ class InfosetWalker private (
       if (child.isSimple) {
         if (!child.isHidden || walkHidden) {
           val simple = child.asInstanceOf[DISimple]
-          outputter.startSimple(simple)
+          doOutputter(outputter.startSimple(simple), "start infoset simple element", simple.erd)
           outputter.endSimple(simple)
```

Comment:
Yep it does. Good catch.

Propogating exceptions isn't something we normally do, and I'm afraid it could cause side effects. For example, if an individual parser catches `Exception` (which they shouldn't do but I think happens in a few places)` then this might get silently ignored, or converted to some unrelated failure diagnostic.

But we are good about parsers not catching SDE exceptions, so I think that's safer to ensure we bubbleup to top and stop the parse.